pkg/icingaredis/telemetry/heartbeat.go

```diff
@@ -146,5 +109,57 @@ func StartHeartbeat(
 	})
 }
 
-// Assert interface compliance.
-var _ json.Marshaler = jsonMetricValue{}
+type GoMetrics struct {
+	names   []string
+	units   []string
+	samples []metrics.Sample
+}
+
+func NewGoMetrics() *GoMetrics {
+	m := &GoMetrics{}
+
+	forbiddenRe := regexp.MustCompile(`\W`)
+
+	for _, d := range metrics.All() {
+		switch d.Kind {
+		case metrics.KindUint64, metrics.KindFloat64:
+			name := "go_" + strings.TrimPrefix(forbiddenRe.ReplaceAllString(d.Name, "_"), "_")
```

Comment:
This is for the case there are <1 _ characters at the beginning, right? Then you could also handle >1 ones by using another trim function.

Reply:
You mean the use of `TrimPrefix()`?

I expect this code to always to always have the same effect as `name := "go_" + forbiddenRe.ReplaceAllString(d.Name[1:], "_")` but using a `strings.Trim*()` function is more robust (in case there's an empty string or one that starts with another character than expected, but both shouldn't happen with the `metrics` package anyways).

But sure, `TrimLeft()` will do the same on all inputs expected from the `metrics` package but would handle more strange inputs correctly (as in not producing `go__something`).